components/inputs/input-percent.js

```diff
@@ -0,0 +1,180 @@
+			/**
+			 * Disables the input
+			 */
+			disabled: { type: Boolean },
+			/**
+			 * Restricts the maximum width of the input box without impacting the width of the label.
+			 */
+			inputWidth: { attribute: 'input-width', type: String },
+			/**
+			 * REQUIRED: Label for the input
+			 */
+			label: { type: String },
+			/**
+			 * Hides the label visually (moves it to the input's "aria-label" attribute)
+			 */
+			labelHidden: { type: Boolean, attribute: 'label-hidden' },
+			/**
+			 * Maximum number of decimal values to show (rounds value up or down).
+			 */
+			maxFractionDigits: { type: Number, attribute: 'max-fraction-digits' },
+			/**
+			 * Minimum number of decimal values to show.
+			 */
+			minFractionDigits: { type: Number, attribute: 'min-fraction-digits' },
+			/**
+			 * Placeholder text
+			 */
+			placeholder: { type: String },
+			/**
+			 * Indicates that a value is required
+			 */
+			required: { type: Boolean },
+			/**
+			 * Text for additional screenreader and mouseover context
+			 */
+			title: { type: String },
+			/**
+			 * Value of the input
+			 */
+			value: { type: Number }
+		};
+	}
+
+	static get styles() {
+		return [ super.styles,
+			css`
+				:host {
+					--d2l-input-text-align: end;
+					display: inline-block;
+					position: relative;
+					width: 100%;
+				}
+				:host([hidden]) {
+					display: none;
+				}
+				:host([disabled]) [slot="right"] {
+					opacity: 0.5;
+				}
+				[slot="right"] {
+					box-sizing: border-box;
+					cursor: default;
+					padding-left: 0.2rem;
+					padding-right: 0.55rem;
+				}
+				:host([dir="rtl"]) [slot="right"] {
+					padding-left: 0.55rem;
+					padding-right: 0.2rem;
+				}
+			`
+		];
+	}
+
+	constructor() {
+		super();
+		this.autofocus = false;
+		this.disabled = false;
+		this.labelHidden = false;
+		this.required = false;
+	}
+
+	firstUpdated(changedProperties) {
+		super.firstUpdated(changedProperties);
+		if (!this.label) {
+			console.warn('d2l-input-percent component requires label text');
+		}
+	}
+
+	render() {
+		return html`
+			<d2l-input-number
+				autocomplete="${ifDefined(this.autocomplete)}"
+				?autofocus="${this.autofocus}"
+				@blur="${this._handleBlur}"
+				@change="${this._handleChange}"
+				?disabled="${this.disabled}"
+				.forceInvalid="${this.invalid}"
+				hide-invalid-icon
+				input-width="${ifDefined(this.inputWidth)}"
+				label="${ifDefined(this.label)}"
+				?label-hidden="${this.labelHidden}"
+				max="100"
+				max-fraction-digits="${ifDefined(this.maxFractionDigits)}"
+				min="0"
+				min-fraction-digits="${ifDefined(this.minFractionDigits)}"
+				name="${ifDefined(this.name)}"
+				?noValidate="${this.noValidate}"
+				placeholder="${ifDefined(this.placeholder)}"
+				?required="${this.required}"
+				?skeleton="${this.skeleton}"
+				title="${ifDefined(this.title)}"
+				value="${ifDefined(this.value)}">
+					<span slot="right" @click="${this._handleSymbolClick}">%</span>
```

Comment:
Not to complicate things, but I just remembered that different locales have different characters and placements of the percent symbol. In reality, I think all the ones we care about are `%`, but the placement does vary slightly. If you grab the `getNumberDescriptor` [out of Intl](https://github.com/BrightspaceUI/intl/blob/master/lib/number.js#L104), you can access the `percentPositivePattern`. Then you can see if the pattern is `{num} %` or `% {num}`. The spaces are also optional, but I don't think you care about them.

We could also add this as a future enhancement.